### PR TITLE
Automatically add related layer from a value relation widget config

### PR DESCRIPTION
**Pull request by 3nids**
and report error if it missing

When setting a layer as being used, if it has dependent layers, they will be automatically used too (in the project config).
In the package layer dialog, error is reported if dependent layers are missing. 

For now, the dependent layers are checked for value relation widget config.
This can be easily extended for other use cases.

ezgif-1-ee2fc731f4e4

### CODE REVIEW

File: qfieldsync/ui/package_dialog.ui [hunk 1]
```diff
@@ -240,10 +240,13 @@
       <item row="0" column="0">
        <widget class="QLabel" name="infoLabel">
         <property name="text">
-         <string>Some layers in this project have not yet been configured. &lt;a href=&quot;configuration&quot;&gt;Configure project now&lt;/a&gt;.</string>
+         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Some issues has been found. &lt;a href=&quot;configuration&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#419cff;&quot;&gt;Configure project now&lt;/span&gt;&lt;/a&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (m-kuhn) [inline qfieldsync/ui/package_dialog.ui:243]:
```suggestion
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Some issues have been found. &lt;a href=&quot;configuration&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#419cff;&quot;&gt;Configure project now&lt;/span&gt;&lt;/a&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```

File: qfieldsync/core/configuration_checkup.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (m-kuhn) [inline qfieldsync/core/configuration_checkup.py:1]: License header

**Comment by m-kuhn** [comment]
@3nids 

What do you think about this?

> I think this is always shown when first setting up a project. Do we want to tell the user that "issues have been found"?

**Comment by 3nids** [comment]
we can reformulate to something closer than the current formulation, or you would prefer to say nothing?

**Comment by m-kuhn** [comment]
Maybe we can show different messages on "issues" or "no configuration done yet"?

**Comment by m-kuhn** [comment]
@3nids do we continue with that?

**Comment by suricactus** [comment]
Nice feature. :+1: Too many conflicts. :-1:  Closing. :handshake:

**Pull request closed without merging**